### PR TITLE
Multiple objective plots

### LINKED ISSUE #51: Objective plots should support multi objective optimization cases
Currently, the objective plot assumes there is always only one objective function.

In the case of multi-objective optimization, I think we should be able to display the objective values with the P10, mean and P90 values for each of the objective functions calculated as well as for the single objective results.

### PR DESCRIPTION
**Pull request by verveerpj**
Plot multiple objective functions. Changed the objective plot to separate multiple objective functions, providing a pull-down to select the desired functions. In addition a plot of the total objective values, the one optimized by everest is shown.

closes #51

### CODE REVIEW

File: everviz/pages/objectives.py [hunk 1]
```diff
@@ -1,41 +1,91 @@
...
 
-def _calc_mean(df):
-    for batch in df["batch"].unique():
-        df.loc[df["batch"] == batch, "Mean"] = df[df["batch"] == batch]["value"].mean()
-    return df
+def _objective_statistics(data):
+    # Aggregate the values over the simulations, using pivot_table, keeping the
+    # batch and function as the multi-index, calculating statistics of the
+    # values over the simulations.
+    statistics = pd.pivot_table(
```
Comment (oyvindeide) [inline everviz/pages/objectives.py:43]: Isn´t this almost identical to the statistics you made for summary keys? Could we use the same function for both?

File: everviz/plugins/summary_plot/summary_plot.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
 from pathlib import Path
 from uuid import uuid4
-from collections import deque
+from itertools import cycle
```
Comment (oyvindeide) [inline everviz/plugins/summary_plot/summary_plot.py:3]: The changes in this file look fine, but can you put them in a different commit, doesn´t quite belong in the same as objectives?

File: everviz/plugins/objectives_plot/objectives_plot.py [hunk 3]
```diff
@@ -71,38 +86,77 @@ def layout(self):
...
+                                x=func_data["batch"],
+                                mode="lines",
+                                marker={"size": 10},
+                                line={"color": color},
+                                name=key,
+                                showlegend=True,
+                            ),
+                        ]
+                    )
+                else:
```
Comment (oyvindeide) [inline everviz/plugins/objectives_plot/objectives_plot.py:145]: The two plugins are now looking quite similar and do the same, they might not be able to be completely merged, but could the parts that overlap be split into shared functions?

File: everviz/plugins/summary_plot/summary_plot.py (hunk 2 above)
Comment (verveerpj) [reply everviz/plugins/summary_plot/summary_plot.py:3]: I can do that, are we ok with putting it in the same pull request? It is just a minor change for consistency.

File: everviz/plugins/objectives_plot/objectives_plot.py (hunk 3 above)
Comment (verveerpj) [reply everviz/plugins/objectives_plot/objectives_plot.py:145]: That is worth a try, but last time I tried this, it was quite a task. So, I would rather leave that for a separate issue/pull request.

File: everviz/pages/objectives.py (hunk 1 above)
Comment (verveerpj) [reply everviz/pages/objectives.py:43]: Yes, it is the same code logic, but they are not quite the same, since the shapes of the input data frames is different. A shared function would be either more complicated and harder to  understand, or only separate out the very basic function calls. I think it is better to leave it as it is.

File: everviz/plugins/summary_plot/summary_plot.py (hunk 2 above)
Comment (oyvindeide) [reply everviz/plugins/summary_plot/summary_plot.py:3]: Yes, the same PR is ok by me.

File: tests/unit/test_objective_plot_data.py [hunk 4]
```diff
@@ -1,45 +1,101 @@
...
 
 
+def test_objective_values_data_frame():
+    """Test for the correct layout and size of the objective values data frame."""
+    objective_values = _objective_values(pd.DataFrame(OBJECTIVES))
+
+    assert set(
+        objective_values.columns
+        == set(["batch", "function", "realization", "simulation"])
+    )
```
Comment (DanSava) [inline tests/unit/test_objective_plot_data.py:34]: Not really sure what this is testing... but the result of
```
set(
        objective_values.columns
        == set(["batch", "function", "realization", "simulation"])
    )
```
is {False} which evaluates to `True`, this seems a bit strange to me 

did you want to check this?
```
assert set(objective_values.columns) == {"batch", "function", "value", "simulation"}
```

Comment (verveerpj) [reply tests/unit/test_objective_plot_data.py:34]: Indeed, good catch!

File: tests/unit/test_objective_plot_data.py [hunk 5]
```diff
@@ -1,45 +1,107 @@
...
+    assert objective_statistics["P10"].to_list() == p10
+    assert objective_statistics["P90"].to_list() == p90
+
+
 def test_insert_p10_p90():
     df = pd.DataFrame(OBJECTIVES)
-    result = _calc_p10_p90(df)
+    result = _objective_statistics(df)
     for header in ["P10", "P90"]:
         assert header in result
```
Comment (DanSava) [inline tests/unit/test_objective_plot_data.py:90]: I think this is already tested as part of the new test `test_objective_statistics_data_frame`

File: everviz/pages/objectives.py [hunk 6]
```diff
@@ -44,6 +94,19 @@ def page_layout(api):
...
         "content": [
             "## Objective function values",
-            {"ObjectivesPlot": {"data_path": sources.objective_values,},},
+            {
+                "ObjectivesPlot": {
+                    "values_file": sources.objective_values,
+                    "statistics_file": sources.objective_statistics,
+                },
+            },
+            "## Minimizer objective function values",
```
Comment (DanSava) [inline everviz/pages/objectives.py:103]: I would rename this to either `Weighted objective` or `Single objective`
